### PR TITLE
support 24bit playback for flac

### PR DESCRIPTION
**Pull request by Wang-Yue**
There's currently a bug in player. Seeking 24bit doesn't work for now.

@flyingmutant  any idea why?

### CODE REVIEW

**Comment by Wang-Yue** [comment]
comment:

1. write_cb performance is slightly worse than the original, which is expected --- previous version just copy 32 bit integer, but this version has to convert 32bit to 24. other bits (8, 16, 32) playback is also slightly worse. but the performance is not a bottleneck, as flac decoding takes much longer time than write_cb. 
2. while write_cb takes a little bit more time, it takes less time for output plugin, and system audio daemon to run (coreaudiod on Mac OS X for instance).
3. seek doesn't work. I don't know why. perhaps we have the wrong read/write position? 
4. On Linux when selecting raw hardware (hw:1:0 for instance), previous version doesn't work since my sound card doesn't support 32 bit. Now it works.

**Comment by mahkoh** [comment]
>There's currently a bug in player. Seeking 24bit doesn't work for now.

Works for me.

**Comment by Wang-Yue** [comment]
@mahkoh what output/input plugin and os version are you using? I want to isolate the problem.

**Comment by mahkoh** [comment]
Pulse 9.0, linux, Flac with your patch applied. I converted an mp3 to 16/24 bit flac and didn't notice a difference in seeking behavior.

**Comment by Wang-Yue** [comment]
OK, I just confirmed it's not a flac plugin bug. 

24bit wav has same issue on my Mac, which I will fix it soon. 

The PR is ready to merge.

File: flac.c [hunk 1]
```diff
@@ -235,50 +234,17 @@ static FLAC__StreamDecoderWriteStatus write_cb(const Dec *dec, const FLAC__Frame
...
+	dest = priv->buf + priv->buf_wpos;
+	for (i = 0; i < frames; i++) {
+		for (ch = 0; ch < channels; ch++) {
+			src = LE32(buf[ch % nch][i] << (bits - depth));
+			src_ptr = (void *)&src;
+			for (j = 0; j < bits / 8; j++) {
+				*dest = *src_ptr;
+				dest++;
+				src_ptr++;
+			}
```
Comment (mahkoh) [inline flac.c]: Why not just memcpy?

Comment (Wang-Yue) [inline flac.c]: done.

**Pull request merged**